farmdata2_modules/fd2_tabs/resources/CustomTableComponent.js

```diff
@@ -62,38 +62,48 @@ let CustomTableComponent = {
     },
     data() {
         return {
-            rowToEdit: null,
+            rowToEditIndex: null,
             indexesToChange: [],
+            editedRowData: {}
+
         }
     },
     methods: {
         editRow: function(index){
-            this.rowToEdit = index
+            this.rowToEditIndex = index
+            this.editedRowData = { 
+                'id': this.rows[index].id,
+                'data': this.rows[index].data
+            }
             this.$emit('edit-clicked')
         },
-        finishRowEdit: function(id, row){
-            this.rowToEdit = null
+        finishRowEdit: function(id){
+            this.rowToEditIndex = null
             let jsonObject = {}
             for(i=0; i < this.indexesToChange.length; i ++){
                 let key = this.headers[this.indexesToChange[i]]
-                jsonObject[key] = row.data[this.indexesToChange[i]]
+                jsonObject[key] = editedRowData.data[this.indexesToChange[i]]
             }
             this.indexesToChange = []
 
+            this.editedRowData = {}
+
             this.$emit('row-edited', jsonObject, id)
         },
         deleteRow: function(id){
             this.$emit('row-deleted', id)
         },
         changedCell: function(itemIndex){
             if(!this.indexesToChange.includes(itemIndex)){
-                this.indexesToChange.push(itemIndex)
+                if(this.editedRowData.data[itemIndex] != this.rows[this.rowToEditIndex].data[itemIndex]) {
```

Comment:
Would it make sense to do this check at the end as well?  That way if something is changed and then changed back to what it was originally it will not be sent as a change?

Reply:
What about adding an else if in this method that checks if the index is already in indexesToChange? Then we can remove it here rather than doing it later on.

Reply:
Yes, I think this function could be restructured to add changes and to also remove indices and data that were "unchanged" (i.e. changed and then changed back).